src/main/java/com/thealgorithms/maths/FibonacciNumber.java

```diff
@@ -0,0 +1,42 @@
+// Created by Pronay Debnath
+// Date:- 1/10/2003
+// Updated FibonnaciNumber.java using golden ratio formula
+// Explanation:- https://www.quickanddirtytips.com/articles/what-is-the-golden-ratio-and-how-is-it-related-to-the-fibonacci-sequence/
+
+import java.util.*;
+
+public class FibonacciNumber<T extends Number> {
```

Comment:
Let's make it a _proper utility class_.

```suggestion
public final class FibonacciNumber<T extends Number> {
    private FibonacciNumber() {
    }
```